crates/miden-testing/src/utils.rs

```diff
@@ -1,8 +1,75 @@
-use miden_lib::transaction::memory;
+use alloc::string::String;
+
+use miden_lib::transaction::{TransactionKernel, memory};
+use miden_objects::{account::AccountId, asset::Asset, note::Note, testing::note::NoteBuilder};
+use rand::rng;
 
 // TEST HELPERS
 // ================================================================================================
 
 pub fn input_note_data_ptr(note_idx: u32) -> memory::MemoryAddress {
     memory::INPUT_NOTE_DATA_SECTION_OFFSET + note_idx * memory::NOTE_MEM_SIZE
 }
+
+/// Creates a note that carries `assets` and a script that moves the assets into the account's
+/// vault.
+///
+/// The created note does not require account authentication and can be consumed by any account.
+pub fn create_transfer_mock_note(sender: AccountId, assets: &[Asset]) -> Note {
```

Comment:
This is one of the functions factored out to create similar scenarios to what was being executed before.